torch_geometric/nn/dense/linear.py

```diff
@@ -215,48 +216,39 @@ def __init__(self, in_channels: int, out_channels: int, num_types: int,
         self.num_types = num_types
         self.is_sorted = is_sorted
         self.kwargs = kwargs
-
-        if torch_geometric.typing.WITH_PYG_LIB:
-            self.lins = None
-            if self.in_channels == -1:
-                self.weight = nn.parameter.UninitializedParameter()
-                self._hook = self.register_forward_pre_hook(
-                    self.initialize_parameters)
-            else:
-                self.weight = torch.nn.Parameter(
-                    torch.Tensor(num_types, in_channels, out_channels))
-            if kwargs.get('bias', True):
-                self.bias = Parameter(torch.Tensor(num_types, out_channels))
-            else:
-                self.register_parameter('bias', None)
+        self.use_segmm: int = -1
+        if self.in_channels == -1:
+            self.weight = nn.parameter.UninitializedParameter()
+            self._hook = self.register_forward_pre_hook(
+                self.initialize_parameters)
+        else:
+            self.weight = torch.nn.Parameter(
+                torch.Tensor(num_types, in_channels, out_channels))
+        if kwargs.get('bias', True):
+            self.bias = Parameter(torch.Tensor(num_types, out_channels))
         else:
-            self.lins = torch.nn.ModuleList([
-                Linear(in_channels, out_channels, **kwargs)
-                for _ in range(num_types)
-            ])
-            self.register_parameter('weight', None)
             self.register_parameter('bias', None)
-
         self.reset_parameters()
 
     def reset_parameters(self):
         r"""Resets all learnable parameters of the module."""
-        if torch_geometric.typing.WITH_PYG_LIB:
-            reset_weight_(self.weight, self.in_channels,
-                          self.kwargs.get('weight_initializer', None))
-            reset_weight_(self.bias, self.in_channels,
-                          self.kwargs.get('bias_initializer', None))
-        else:
-            for lin in self.lins:
-                lin.reset_parameters()
+        reset_weight_(self.weight, self.in_channels,
+                      self.kwargs.get('weight_initializer', None))
+        reset_weight_(self.bias, self.in_channels,
+                      self.kwargs.get('bias_initializer', None))
 
     def forward(self, x: Tensor, type_vec: Tensor) -> Tensor:
         r"""
         Args:
             x (torch.Tensor): The input features.
             type_vec (torch.Tensor): A vector that maps each entry to a type.
         """
-        if torch_geometric.typing.WITH_PYG_LIB:
+        if x.size(0) == 1:
+            # for single vector inference don't use segmm
+            self.use_segmm = 0
```

Comment:
Do you want to reset this to its original value at the end. If not if the user passes larger values later, `use_segmm` will be stuck at 0.
Also what if we just don't have this logic, does it cause bugs? 
```suggestion
```

Reply:
i think we can leave this off tbh, the hueristic should choose the right choice